src/tiled/objectiddialog.cpp

```diff
@@ -0,0 +1,176 @@
+/*
+ * objectiddialog.cpp
+ * Copyright 2016, Thorbjørn Lindeijer <bjorn@lindeijer.nl>
+ *
+ * This file is part of Tiled.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "objectiddialog.h"
+#include "ui_objectiddialog.h"
+
+#include "documentmanager.h"
+#include "mapdocument.h"
+#include "map.h"
+#include "objectgroup.h"
+#include "mapobject.h"
+#include "utils.h"
+
+#include <QList>
+#include <QLineEdit>
+#include <QTableWidget>
+#include <QTableWidgetItem>
+#include <QTableWidgetSelectionRange>
+#include <QPushButton>
+#include <QHeaderView>
+
+namespace Tiled {
+namespace Internal {
+
+
+ObjectIdDialog::ObjectIdDialog(QWidget *parent)
+    : QDialog(parent)
+    , mUi(new Ui::ObjectIdDialog)
+    , mId(0)
+{
+    mUi->setupUi(this);
+    setWindowFlags(windowFlags() & ~Qt::WindowContextHelpButtonHint);
+
+    Utils::restoreGeometry(this);
+
+    if (MapDocument *document = DocumentManager::instance()->currentDocument()) {
+        QTableWidget *tableWidget = mUi->tableWidget;
```

Comment:
Rather than using a `QTabeWidget` and copying all objects into it, did you try using a `QTreeView` and re-using the `MapObjectModel`? When combined with a `QSortFilterProxyModel`, you can still have sorting and filtering work. Also, you could add an ID column to `MapObjectModel`, but hide it in the `ObjectsView` (in `objectsdock.cpp`).